File: shared-libraries/java/localpos-common/src/main/java/com/pds/localpos/common/util/CorrelationIdHolder.java
package com.pds.localpos.common.util;

import lombok.experimental.UtilityClass;

@UtilityClass
public class CorrelationIdHolder {

    private static final ThreadLocal<String> correlationId = new ThreadLocal<>();

    public void set(String id) {
        correlationId.set(id);
    }

    public String get() {
        return correlationId.get();
    }

    public void clear() {
        correlationId.remove();
    }
}

File: shared-libraries/kotlin/localpos-common-kt/bin/main/com/pds/localpos/common/util/CorrelationIdHolder.kt
package com.pds.localpos.common.util

object CorrelationIdHolder {
    private val correlationId = ThreadLocal<String>()

    fun set(id: String) {
        correlationId.set(id)
    }

    fun get(): String? {
        return correlationId.get()
    }

    fun clear() {
        correlationId.remove()
    }
}
